reliability/tasks/Task.py

```diff
@@ -101,45 +152,112 @@ def execute(self):
                 if len(all_apps.apps) > 0:
                     apps = list(all_apps.apps.keys())
                     apps_to_build = self.get_targets(apps, self.get_percent())
+                    # prepare parameters list - app name and kubeconfig - for app.build
+                    app_build_args = []
                     for app_to_build in apps_to_build:
-                        all_apps.apps[app_to_build].build()
+                        try:
+                            project = all_apps.apps[app_to_build].project
+                            user = all_projects.projects[project].user
+                            kubeconfig = global_data.kubeconfigs[user]
+                            app_build_args.append((app_to_build, kubeconfig))
+                        except KeyError:
+                            self.logger.info(f"Projet {project_to_modify} no longer exists in projects list.")
+                    # add apps concurrently
+                    with ThreadPoolExecutor(max_workers=workers) as executor:
+                        results = executor.map(lambda t: all_apps.apps[t[0]].build(t[1]), app_build_args)
+                        for result in results:
+                            self.logger.info(result)
+
             elif action == "scaleUp":
                 self.logger.debug("ScaleUp apps")
-                apps = list(all_apps.apps.keys())
-                apps_to_scale = self.get_targets(apps, self.get_percent())
-                for app_to_scale in apps_to_scale:
-                    all_apps.apps[app_to_scale].scale_up()
+                if len(all_apps.apps) > 0:
+                    apps = list(all_apps.apps.keys())
+                    apps_to_scale = self.get_targets(apps, self.get_percent())
+                    # prepare parameters list - app name and kubeconfig - for app.scale_up
+                    app_scale_args = []
+                    for app_to_scale in apps_to_scale:
+                        try:
+                            project = all_apps.apps[app_to_scale].project
+                            user = all_projects.projects[project].user
+                            kubeconfig = global_data.kubeconfigs[user]
+                            app_scale_args.append((app_to_scale, kubeconfig))
+                        except KeyError:
+                            self.logger.info(f"Projet {project_to_modify} no longer exists in projects list.")
```

Comment:
Same logging issues here: don't think the project_to_modify variable will be set here, need to change to just project or change the logging line.

Reply:
Thank you for the catching!